src/cli/serve/deprecated_config.js

```diff
@@ -0,0 +1,16 @@
+import { forOwn, has, noop } from 'lodash';
+
+// deprecated settings are still allowed, but will be removed at a later time. They
+// are checked for after the config object is prepared and known, so legacySettings
+// will have already been transformed.
+export const deprecatedSettings = new Map([
+  [['server', 'xsrf', 'token'], 'server.xsrf.token is deprecated. It is no longer used when providing xsrf protection.']
```

Comment:
What are the drawbacks to that? I do think this should work for any valid way to configure a key.

If that means that you must pass a fully normalized/merged config, then I think that's completely fine, but it should probably throw an exception if it were given anything else rather than silently ignoring them.